query.py

```diff
@@ -152,11 +158,8 @@ def create_hubmap_metadata_df(
                     )
                     continue
 
-                # Dynamically collect all URLs from dataset class
-                urls = {
-                    f"{field}_url": getattr(dataset_urls, field) or ""
-                    for field in dataset_class.__annotations__
-                }
+                # Convert the dataset_class instance to a *_url dictionary
+                urls = {f"{k}_url": v or "" for k, v in dataset_urls._asdict().items()}
```

Comment:
I think this could be

```python

from dataclasses import asdict
```

and then using that instead of an underscore method.

Reply:
True, but `asdict()` should be called on dataclass instances, but `BulkseqDataset` and `scRNAseqDataset` are of `NamedTuple`. Would you like me to convert the classes to `@dataclass`?

Reply:
Ah no sorry. I misremembered. Fine then.